src/classes/BDI_DataImportService.cls

```diff
@@ -525,6 +529,20 @@ global with sharing class BDI_DataImportService {
                 populatedDataImportFields[i] = populatedDataImportFields[i].toLowerCase();
             }
 
+            this.flsValidator =
+                    new FieldLevelSecurityValidator(
+                            dataImports,
+                            fieldMapper.getTargetObjectsAndFieldsByDataImportField());
+
+            //From this point forward, only operate on the Data Import records that passed FLS
+            //validation.  Records that did not pass FLS validation will be updated with their
+            //respective FLS exceptions at the end of this block
+            if (isDryRun) {
+                this.listDI = flsValidator.getDataImportsWithAccessibleTargetFields();
+            } else {
```

Comment:
Do we really check different levels of access for dry run vs 'real' processing? You don't NEED write access to dry run, but would it be better to advertise the FLS error that you WOULD get in processing from dry run?